### PR TITLE
Day20

### PR DESCRIPTION
**Pull request by JustineSara**
it's again a horrible mess with stuff half-written and un-used ...

### CODE REVIEW

File: src/t/d20.clj [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  [walls E xmax ymax]
+  (fn [S]
+    (loop [p [S]
+           N 0]
+        (let [np (mapcat
+                   (fn [[x y]] [[(inc x) y] [(dec x) y] [x (inc y)] [x (dec y)]])
+                   p)
+              np (->> np
+                      (filter (fn [[x y]] (inside? x y xmax ymax)))
+                      (filter #(not (contains? walls %))))]
```
Comment (gaverhae) [inline src/t/d20.clj:65]: This should work as well:
```suggestion
                      (remove walls)]
```

File: src/t/d20.clj [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+(defn def-find-shortest-path
+  [walls E xmax ymax]
+  (fn [S]
+    (loop [p [S]
+           N 0]
+        (let [np (mapcat
+                   (fn [[x y]] [[(inc x) y] [(dec x) y] [x (inc y)] [x (dec y)]])
+                   p)
+              np (->> np
+                      (filter (fn [[x y]] (inside? x y xmax ymax)))
```
Comment (gaverhae) [inline src/t/d20.clj:64]: If you redefine `inside?` to be something like:

```clojure
(defn inside?
  [width height]
  (let [xmax (dec width)
        ymax (dec height)]
  (fn [[x y]]
    (and (<= 0 x xmax) (<= 0 y ymax)))))
```

you can write:

```suggestion
                      (filter (inside? xmax ymax))
```

**Review by gaverhae — APPROVED** [review]

File: src/t/d20.clj (hunk 1 above)
Comment (JustineSara) [reply src/t/d20.clj:65]: oooh!

File: src/t/d20.clj (hunk 2 above)
Comment (JustineSara) [reply src/t/d20.clj:64]: yes, I am slowly starting to use `fn` but it is in progress

**Pull request merged**